ee/tabby-webserver/src/schema/gitlab_repository_provider.rs

```diff
@@ -0,0 +1,136 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use juniper::{GraphQLInputObject, GraphQLObject, ID};
+use lazy_static::lazy_static;
+use regex::Regex;
+use validator::Validate;
+
+use super::Context;
+use crate::{juniper::relay::NodeType, schema::Result};
+
+lazy_static! {
+    static ref GITLAB_REPOSITORY_PROVIDER_NAME_REGEX: Regex = Regex::new("^[\\w-]+$").unwrap();
```

Comment:
consider extract this regex to some where (e.g schema/constants.rs) and reuse it between:
1. Github display name
2. Gitlab display name
3. Git name